diff -wruN batteries-1.3.0.orig/Makefile batteries-1.3.0/Makefile
--- batteries-1.3.0.orig/Makefile	2011-07-05 04:16:51.000000000 +0300
+++ batteries-1.3.0/Makefile	2011-07-05 04:17:08.000000000 +0300
@@ -57,7 +57,7 @@
 	$(OCAMLBUILD) batteries.docdir/index.html
 	test -e apidocs || ln -s _build/batteries.docdir apidocs
 
-install: all
+install:
 	ocamlfind install $(OCAMLFIND_DEST) estring \
 		libs/estring/META \
 		_build/libs/estring/*.cmo \
@@ -70,7 +70,7 @@
 	ocamlfind remove $(OCAMLFIND_DEST) $(NAME)
 	rm -rf $(DOCROOT)
 
-install-doc: doc
+install-doc:
 	mkdir -p $(DOCROOT)
 	cp -r doc/batteries/* $(DOCROOT)
 # deal with symlink that will break
@@ -105,6 +105,9 @@
 ifeq ($(CAMVER),0.8.2)
 	CAMFIX=camomile82
 endif
+ifeq ($(CAMVER),0.8.3)
+	CAMFIX=camomile83
+endif
 ifeq ($(CAMVER),0.8.1)
 	CAMFIX=camomile81
 endif
@@ -127,6 +130,10 @@
 camomile82:
 	cp -f src/batCamomile-0.8.2.ml src/batCamomile.ml
 
+#replace batcamomile with a version appropriate for camomile 0.8.3
+camomile83:
+	cp -f src/batCamomile-0.8.2.ml src/batCamomile.ml
+
 #replace batcamomile with a version appropriate for camomile 0.8.1
 camomile81:
 	cp -f src/batCamomile-0.8.1.ml src/batCamomile.ml
